src/protocolsupport/protocol/typeremapper/itemstack/complex/ItemStackComplexRemapperRegistry.java

```diff
@@ -111,29 +115,31 @@ protected static void register(
 	static {
 		registerFromClient(Material.FILLED_MAP, new MapFromLegacyIdComplexRemapper(), ProtocolVersionsHelper.BEFORE_1_13);
 		registerFromClient(Material.POTION, new PotionFromLegacyIdComplexRemapper(), ProtocolVersionsHelper.BEFORE_1_9);
-		EnchantFromLegacyIdComplexRemapper enchantfromlegacyid = new EnchantFromLegacyIdComplexRemapper();
-		DisplayNameFromLegacyTextComplexRemapper dnamefromlegacytext = new DisplayNameFromLegacyTextComplexRemapper();
-		EnchantFromPEEnchantRemapper frompeenchantremapper = new EnchantFromPEEnchantRemapper();
+		ItemStackComplexRemapper enchantfromlegacyid 	= new EnchantFromLegacyIdComplexRemapper();
+		ItemStackComplexRemapper dnamefromlegacytext 	= new DisplayNameFromLegacyTextComplexRemapper();
+		ItemStackComplexRemapper frompeenchantremapper 	= new EnchantFromPEEnchantRemapper();
+		ItemStackComplexRemapper pestashremapper 		= new NBTUnStashRemapper();
+		BookPagesFromPERemapper frompebook 				= new BookPagesFromPERemapper();
+		PotionFromPEIdRemapper frompepotion 			= new PotionFromPEIdRemapper();
+		LeatherArmorFromPERemapper frompeleatherarmor 	= new LeatherArmorFromPERemapper();
 		Arrays.stream(Material.values())
 		.filter(Material::isItem)
 		.forEach(material -> {
-			registerFromClient(material, enchantfromlegacyid, ProtocolVersionsHelper.BEFORE_1_13);
+			//TODO: do we need these anymore? backwards mapping done with stashed NBT now
```

Comment:
Is backwards nbt also in the creative inventory? I don't think so. Now we could add it to the creative-inventory items, but I think that be too much work to keep up.

Reply:
its not :9 so once we get items creative items back in the creative inventory with NBT, we'll need to make sure everything is lined up correctly.